tests/models/test_rate_limited.py

```diff
@@ -0,0 +1,639 @@
+from __future__ import annotations
+
+from collections.abc import AsyncIterator
+from contextlib import asynccontextmanager
+from datetime import datetime, timezone
+
+import pytest
+from tenacity import AsyncRetrying, RetryError, retry_if_exception_type, stop_after_attempt, wait_fixed
+from throttled.asyncio import Throttled, rate_limiter
+from throttled.asyncio.store import MemoryStore
+
+from pydantic_ai.messages import (
+    ModelMessage,
+    ModelRequest,
+    ModelResponse,
+    ModelResponseStreamEvent,
+    PartDeltaEvent,
+    PartStartEvent,
+    SystemPromptPart,
+    TextPart,
+    TextPartDelta,
+    UserPromptPart,
+)
+from pydantic_ai.models import Model, ModelRequestParameters, StreamedResponse
+from pydantic_ai.models.rate_limited import RateLimitedModel
+from pydantic_ai.settings import ModelSettings
+from pydantic_ai.usage import Usage
+
+
+class FailingModel(Model):
```

Comment:
We already do some testing with fake failing models in `tests/models/test_fallback.py`. Could we borrow the approach taken there using `FunctionModel`?